components/AddDogForm/index.js

```diff
@@ -0,0 +1,191 @@
+import Link from "next/link";
+import styled from "styled-components";
+import { useForm } from "react-hook-form";
+
+export default function AddDogForm() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const onSubmit = (data) => console.log(data);
```

Comment:
In your PR description, you mention "saving form data", but for now, your form is just logging the data to console. You should try to fully implement the saving functionality (e.g. make your static dog data a global state and add the saved entries to it) and avoid any `console.logs` revealing your data structure.